orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -350,30 +380,38 @@ Status FromPropertyBag(const PropertyBag& property_bag, flatbuffers::FlatBufferB
     return Status::OK();
   }
 
+  InlinedVector<std::string> sorted_property_names;
+  sorted_property_names.reserve(property_bag.Size());
+  for ([[maybe_unused]] const auto& [name, value] : property_bag) {
+    sorted_property_names.push_back(name);
+  }
+  std::sort(sorted_property_names.begin(), sorted_property_names.end());
+
   std::vector<flatbuffers::Offset<fbs::IntProperty>> ints;
   std::vector<flatbuffers::Offset<fbs::FloatProperty>> floats;
   std::vector<flatbuffers::Offset<fbs::StringProperty>> strings;
-  for (const auto& [name, value] : property_bag) {
+  for (const auto& name : sorted_property_names) {
+    const auto& value = property_bag.GetProperty<PropertyDataType>(name);
     const auto fbs_property_name = builder.CreateString(name);
     if (std::holds_alternative<int64_t>(value)) {
       fbs::IntPropertyBuilder int_property_builder(builder);
       int_property_builder.add_name(fbs_property_name);
       int_property_builder.add_value(std::get<int64_t>(value));
       flatbuffers::Offset<fbs::IntProperty> property = int_property_builder.Finish();
-      ints.emplace_back(property);
+      ints.emplace_back(std::move(property));
```

Comment:
nit: Offset is an effectively an integer so I think move is unnecessary here. sorry if my previous comment was too broad. I'll limit this comment to this for loop :)